{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[' Trey Ideker, Ph. D. is Professor of Medicine at the University of California at San Diego. He serves as Division Chief of Medical Genetics and Director of the National Resource for Network Biology, as well as being Adjunct Professor of Bioengineering and Computer Science and Member of the Moores UCSD Cancer Center. Ideker received Bachelor’s and Master’s degrees from MIT in Electrical Engineering and Computer Science and his Ph.D. from the University of Washington in Molecular Biology under the supervision of Dr. Leroy Hood. He is a pioneer in assembling genome-scale measurements to construct network models of cellular processes and disease. His recent research activities include assembly of networks governing the response to DNA damage; development of the Cytoscape and NetworkBLAST software packages for biological network visualization and cross-species network comparison; and methods for identifying network-based biomarkers in development and disease. Ideker serves on the Editorial Boards for Bioinformatics and PLoS Computational Biology, is on the Scientific Advisory Boards of the Sanford-Burnham Medical Research Institute and the Institute for Systems Biology, and is a regular consultant for companies such as Monsanto and Mendel Biotechnology. He was named one of the Top 10 Innovators of 2006 by Technology Review magazine and was the recipient of the 2009 Overton Prize from the International Society for Computational Biology. His work has been featured in news outlets such as The Scientist, the San Diego Union Tribune, Forbes magazine and the New York Times. ']\n",
      "**********************************************\n",
      "[' Timothy R. Hughes is a Professor at the Donnelly CCBR at the University of Toronto. He studied engineering and music at the University of Iowa, and received his Ph.D. in Cell and Molecular Biology from Baylor College of Medicine, working on telomere replication proteins with Vicki Lundblad. He did his postdoctoral work at Rosetta Inpharmatics (now Merck) working on microarray technology and its applications, including the development of ink-jet arrays now available from Agilent. Since moving to Toronto in 2001, Dr. Hughes has been the recipient of a Canada Research Chair in Genome Biology, the Ontario Premier’s Research Excellence Award, the Terry Fox Young Investigator award, and an HHMI foreign scholarship. He has authored or co-authored over 100 manuscripts, and is a scholar of the Canadian Institutes For Advanced Research. His laboratory has worked in gene regulation, systems biology, functional genomics, RNA processing, and genome sequencing.']\n",
      "**********************************************\n",
      "[' Brenda Andrews is Professor and Chair of the Banting & Best Department of Medical Research within the Faculty of Medicine at the University of Toronto, where she holds the Charles H Best Chair in Medical Research. She is also Director of the Terrence Donnelly Center for Cellular and Biomolecular Research (the Donnelly Centre), an interdisciplinary biomedical research institute with a focus on technology development for post-genome biology, functional genomics, systems & computational biology and bioengineering. After receiving her PhD in Medical Biophysics from the University of Toronto, Dr. Andrews obtained her early training in genetics with the late Dr. Ira Herskowitz at the University of California San Francisco. In 1991, Dr. Andrews was recruited to the Department of Medical Genetics (now Molecular Genetics) at the University of Toronto. She became Chair of the Department in 1999, a position she held for 5 years before assuming her current positions. Dr. Andrews’ current research interests analysis of genetic interaction networks in budding yeast, using automated genetics platforms that include high content microscopy for systematic analysis of cell biological phenotypes. Specific interests in the Andrews lab include mechanisms of cell cycle control, control of cell function by kinases and other enzymes and the regulation of cell polarity and morphogenesis. Her research is currently funded by the CIHR, the National Institutes of Health, the Ontario Research Fund, the Canadian Foundation for Innovation and the Canadian Institute for Advanced Research (CIFAR).Dr. Andrews is a Fellow of the Royal Society of Canada, Fellow of the American Association for the Advancement of Science, a Fellow of the American Academy of Microbiology and Director of the Genetic Networks Program of the CIFAR']\n",
      "**********************************************\n",
      "[' Dr. Peter Jackson is Professor in the Baxter Laboratory for Stem Cell Research at Stanford, and formerly Director and Staff Scientist at Genentech. His laboratory develops cancer drug targets and pioneered proteomic network building to identify diagnostics and targets in human disease. Their discovery of new disease genes and signaling pathways in the primary cilium, a rediscovered signaling organelle, helps explain neurological and kidney disease, obesity, retinal degeneration, and cancer. ']\n",
      "**********************************************\n",
      "[' Galit Lahav received her PhD in 2001 from the Department of Biology in the Technion, Israel Institute of Technology. In 2003, she completed her postdoctoral fellowship at the Weizmann Institute of Science in Israel. She then spent a year at Harvard’s Bauer Center for Genomics Research, and in the fall of 2004, joined the Department of Systems Biology at Harvard Medical School. Her research group combines experimental and computational approaches to understand how drugs act on different cell types and organs, and to gain insight into the reasons why different cells and people respond differently to specific drugs. Lahav is also a dedicated mentor to new faculty and highly committed to furthering the advancement of women in science']\n",
      "**********************************************\n",
      "[' Bernhard Palsson is the Galletti Professor of Bioengineering and the Principal Investigator of the Systems Biology Research Group in the Department of Bioengineering at the University of California, San Diego. Dr. Palsson has co-authored more than 360 peer-reviewed research articles and has authored three textbooks, with one more in preparation. His research includes the development of methods to analyze metabolic dynamics (flux-balance analysis, and modal analysis), and the formulation of complete models of selected cells (the red blood cell, E. coli, hybridoma, and several human pathogens). He sits on the editorial broad of several leading peer-reviewed microbiology, bioengineering, and biotechnology journals. He previously held a faculty position at the University of Michigan for 11 years and was named the G.G. Brown Associate Professor at Michigan in 1989, a Fulbright fellow in 1995, and an Ib Henriksen Fellow in 1996. He is the author of 40 U.S. patents, the co-founder of several biotechnology companies, and holds several major biotechnology awards. He received his PhD in Chemical Engineering from the University of Wisconsin, Madison. Dr. Palsson is a member of the National Academy of Engineering and is a Fellow of both the AAAS and the AAM']\n",
      "**********************************************\n",
      "[' Olivier Pourquié, Ph.D. is currently Professor at Strasbourg University Medical School (France). He was the director of the Institute for Genetics and Molecular and Cellular Biology in Strasbourg from 2009 to 2012 and before he was a Howard Hughes Medical Institute Investigator at the Stowers Institute for Medical Research in Kansas City, USA. His lab provided the first evidence of the existence of a molecular oscillator -the segmentation clock- associated to the rhythmic production of vertebral precursors (the somites) in the embryo. This discovery was listed as one of 25 milestones in Developmental Biology in the 20th century by Nature magazine. The work from the Pourquié laboratory has imposed vertebrate segmentation as a novel paradigm to study spatiotemporal regulation of signaling in development. Together, his discoveries have had important consequences for our understanding of the patterning of the vertebrate embryonic axis and provided a conceptual framework to explain human spine malformations such as congenital scoliosis. Pr Pourquié is an elected EMBO and Academia Europea member and is the recipient of several prestigious awards including the Lounsbery Grand Prize of the French and American Academy of Sciences, the Allianz Grand Prize of the French Academy of Sciences and a European Research Council Advanced grant']\n",
      "**********************************************\n",
      "[' Ziv Bar-Joseph is an Associate Professor in the Lane Center for Computational Biology and the Machine Learning Department at the School of Computer Science at Carnegie Mellon University. His work focuses on the analysis and integration of static and temporal high throughput biological data for systems biology and on bi-directional studies linking information processing in biological and computational systems. Dr. Bar-Joseph has been the co-chair of the RECOMB meeting on Regulatory Networks and Systems Biology and is currently an associate editor for Bioinformatics. He received his Ph.D. from the MIT in 2003. He was the recipient of the DIMACS-Celera Genomics Graduate Student Award in Computational Molecular Biology, the NSF CAREER award and the 2012 Overton prize in computational biology. ']\n",
      "**********************************************\n",
      "[' Emmanouil Dermitzakis is currently a Louis-Jeantet Professor of Genetics in the Department of Genetic Medicine and Development of the University of Geneva Medical School. He is a member of the executive board of the Institute of Genetics and Genomics in Geneva (iGE3) and is also an affiliated Faculty member at the Biomedical Research Foundation of the Academy of Athens in Greece. His current research focuses on the genetic basis of cellular phenotypes and complex traits. He has authored and coauthored more than 100 papers in peer-reviewed journals and many of them in journals such as Nature, Science and Nature Genetics. His papers have been cited more than 20000 times and his H-index is 49. His research is supported by the Louis-Jeantet Foundation, the Wellcome Trust, the Swiss National Science Foundation, the European Commission, the Juvenile Diabetes Foundation and the US National Institutes of Health (NIH). He is also the recipient of a European Research Council (ERC) grant. He has been invited to give talks and keynote lectures in most of the prestigious genetics meeting and is the organizer of training courses including the Wellcome Trust HapMap course and founder and organizer of the Leena Peltonen School of Human Genomics. He is currently an analysis co-chair in the GTex project and has served as an analysis co-chair in the pilot phase of the ENCODE (ENCyclopedia Of Dna Elements) consortium and member of the analysis group of the Mouse Genome Sequencing Consortium and the International HapMap project. He had a leading analysis role in the HapMap3 project and is a member of the analysis group of the 1000 genomes project. He has served in the Board of Reviewing Editors of Science (2006-2011), and as a Senior Editor in PLoS Genetics (2006-2012) and is currently a member of the Board of Reviewing editors for the new scientific journal eLIFE. He is also in the advisory board of DNAnexus. ']\n",
      "**********************************************\n",
      "[' Brenton R. Graveley obtained hi B.A. manga cum laude in 1991 from the Department of Molecular, Cellular, and Developmental Biology at the University of Colorado, Boulder where he performed research with David M. Prescott. He conducted his Ph.D. Research under the guidance of Gregory M. Gilmartin in the Department of Microbiology and Molecular Genetics at the University of Vermont and graduated in 1996. He was a Jane Coffin Childs postdoctoral fellow from 1996-1999 with Tom Maniatis in the Department of Molecular and Cellular Biology at Harvard University. Since 1999, he has been a faculty member in the Department of Genetics and Developmental Biology at the University of Connecticut Health Center in Farmington, CT where he is currently a Professor. He is also Associate Director of the Institute for Systems Genomics at the University of Connecticut, a Member of the Board of Directors for the RNA Society, and an Editor of the journal RNA. ']\n",
      "**********************************************\n",
      "['  Jason Lieb is a Professor of Molecular Biology at the Lewis-Sigler Institute for Integrative Genomics at Princeton University. Prior to coming to Princeton, Lieb was the Beverley W. Long Chapin Distinguished Professor of Biology in the UNC College of Arts and Sciences, and the Director of the Carolina Center for Genome Sciences. Research projects in the Lieb lab are united by the scientific goal of understanding relationships between chromatin, transcription factor targeting, and gene expression. In particular, Lieb studies how information is encoded and dynamically utilized in living eukaryotic genomes, focusing on those areas of the genome that serve to regulate chromosomal functions, including transcription, DNA replication and repair, recombination, and chromosome segregation. Lieb came to UNC in 2002 after completing a postdoctoral fellowship at Stanford University. He earned his undergraduate degree in biology at the University of North Carolina at Chapel Hill and his PhD in genetics at the University of California at Berkeley. ']\n",
      "**********************************************\n",
      "[' Aviv Regev, a computational biologist, joined the Broad Institute as a core member and MIT as a faculty member in 2006. Regev’s research centers on understanding how complex molecular networks function and evolve in the face of genetic and environ¬mental changes, over time — scales ranging from minutes to millions of years. Prior to joining the Broad Institute, Regev was a fellow at the Bauer Center for Genomics Research at Harvard University, where she developed new approaches to the reconstruction of regulatory networks and modules from genomic data. Regev also worked for several years in the biotech industry in Israel, where she established and directed a bioinfor¬matics research and development team at QBI, a functional genomics company. Regev is also an associate professor in the Department of Biology at MIT and director of the Klarman Cell Observatory at the Broad. She was named an Early Career Scientist of the Howard Hughes Medical Institute in 2009. She is a recipient of the NIH Director’s Pioneer Award, a Sloan fellowship from the Sloan Foundation, and the Overton Prize from the International Society for Computational Biology. Regev received her M.Sc. from Tel Aviv University, studying biology, computer science, and mathematics in the Interdisciplinary Program for the Fostering of Excellence, where she did research in both theoretical biology (on the evolution of development) and experimental biology (on genomic instability). She received her Ph.D. in computational biology from Tel Aviv University']\n",
      "**********************************************\n",
      "[' Professor Jussi Taipale earned his Ph.D. from the University of Helsinki in 1996. He initially continued his postdoctoral work at the University of Helsinki and later at Johns Hopkins University (Baltimore, MD, USA). Since 2003, he has headed an independent research laboratory focusing on systems biology of growth control and cancer. He has published 63 articles of which thirteen are in the most prestigious scientific journals (Nature, Science, Cell), won numerous awards and grants (e.g., Anders Jahre Prize for Young Researchers, EMBO Young Investigator, ERC Advanced Grant), and is internationally recognized as a leader in the field of genomics and systems biology. The Taipale group’s main expertise is on high-throughput screening using cDNA (Varjosalo et al. Cell 2008) and RNA interference (Björklund et al. Nature 2006), and computational and experimental methods to identify causative regulatory mutations in non-protein coding DNA and to analyze genetic networks (see Jolma et al., Cell 2013; Yan et al., Cell 2013). In addition, Taipale group has extensive expertise on mouse models of gene and regulatory region function (see Dumont et al., Science 1998; Ma et al., Cell 2002; Hallikas et al. Cell 2006; Sur et al., Science 2012). ']\n",
      "**********************************************\n",
      " \n",
      "\t\tPrint\t\n",
      "\n",
      "\n",
      "\n",
      "Trey Ideker\n",
      "\n",
      "Timothy R. Hughes\n",
      "\n",
      "\n",
      "Brenda Andrews\n",
      "\n",
      "Levi A. Garraway\n",
      "\n",
      "\n",
      "Peter Jackson\n",
      "\n",
      "Galit Lahav\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Bernhard Palsson\n",
      "\n",
      "\n",
      "Olivier Pourquié\n",
      "\n",
      "Peter Sorger\n",
      "\n",
      "Ziv Bar-Joseph\n",
      "\n",
      "\n",
      "Emmanouil (Manolis) Dermitzakis\n",
      "\n",
      "Brenton R. Graveley\n",
      "\n",
      "Jason D. Lieb\n",
      "\n",
      "Aviv Regev\n",
      "\n",
      "Jussi Taipale\n",
      " [top]\n"
     ]
    }
   ],
   "source": [
    "#!/usr/bin/env python3\n",
    "# -*- coding: utf-8 -*-\n",
    "#import scrapy\n",
    "import re\n",
    "from urllib.request import urlopen\n",
    "from bs4 import BeautifulSoup\n",
    "#1.because of there is no universial structrue, we need to write one crawler for each webpage\n",
    "#2.even by doing this,some webpages aare difficult to get the bios and names corresponding\n",
    "url=\"https://www.iscb.org/recomb-regsysgen2013-program/recomb-regsysgen2013-keynote-spk-ab-bio\"\n",
    "response=urlopen(url)\n",
    "soup=BeautifulSoup(response.read(),'html.parser',from_encoding='utf-8')\n",
    "nodes=soup.find_all('p')\n",
    "str1=\"\"\n",
    "for node in nodes:\n",
    "    \n",
    "    temp_text=node.get_text()\n",
    "    temp_text=temp_text.split(\"................\")\n",
    "    for text in temp_text:\n",
    "#         print(text)\n",
    "#         print(\"################################################\")\n",
    "         p=re.findall(r'Biography:(.*)',text)\n",
    "         if(len(p)>0):\n",
    "             print(p)\n",
    "             print(\"**********************************************\")\n",
    "    \n",
    "   \n",
    "div_tag=soup.find_all(\"div\",{\"class\":\"item-page\"})\n",
    "for tag in div_tag:\n",
    "    a_tags=tag.find_all(\"a\")\n",
    "    for a in a_tags:\n",
    "        print(a.get_text())\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
